src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -615,20 +611,68 @@ def launch_fluent(
 
     del start_container
 
-    if start_watchdog is None and cleanup_on_exit:
+    if additional_arguments is None:
+        additional_arguments = ""
+    elif fluent_launch_mode == LaunchMode.PIM:
+        raise ValueError(
```

Comment:
these are normally `TypeError`. 
```
>>> def foo(): pass
...
>>>
>>> foo(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() takes 0 positional arguments but 1 was given
>>> foo(x=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() got an unexpected keyword argument 'x'
>>>
```

Reply:
In this case the "type" (`NoneType` or `str`) is correct as the function does accept and expect that variable with those types. However, with this combination of arguments, the "value" is incorrect as it needs to be `None`. I think `ValueError` makes a bit more sense than `TypeError`.

Googling this, most people seem to suggest `ValueError` as well, first Google links that show up here:
https://stackoverflow.com/questions/256222/which-exception-should-i-raise-on-bad-illegal-argument-combinations-in-python

https://www.adamsmith.haus/python/answers/how-to-raise-an-exception-for-illegal-argument-combinations-in-python

Reply:
Okay, that makes sense, yes.